BoxContentSDK/BoxContentSDK/Operations/BOXStreamOperation.m

```diff
@@ -144,12 +144,12 @@ - (void)abortWithError:(NSError *)error
 
 #pragma mark - BOXURLSessionTaskDelegate
 
-- (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response error:(NSError *)error
+- (void)processIntermediateResponse:(NSURLResponse *)response
 {
-    [super sessionTask:sessionTask didFinishWithResponse:response error:error];
-
-    if (error != nil &&
-        error.code == BOXContentSDKAPIErrorAccepted) {
+    [super processIntermediateResponse:response];
+    
+    if (self.error != nil &&
+        self.error.code == BOXContentSDKAPIErrorAccepted) {
```

Comment:
if we want BOXStreamOperation to behave more like fore-ground download to report on progress, we want similar setup as BOXAIDataOperation. We can override createSessionTask to call [self.session.urlSessionManager createNonBackgroundDownloadTaskWithRequest:self.APIRequest taskDelegate:self];
and BOXStreamOperation inherits BOXURLSessionDownloadTaskDelegate